battle-service/src/main/java/by/ibondarau/tictactoe/battleservice/checker/impl/GameCheckerImpl.java

```diff
@@ -0,0 +1,40 @@
+package by.ibondarau.tictactoe.battleservice.checker.impl;
+
+import by.ibondarau.tictactoe.battleservice.checker.GameChecker;
+import by.ibondarau.tictactoe.battleservice.model.Move;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Component("GameChecker")
+public class GameCheckerImpl implements GameChecker {
+    @Override
+    public boolean checkWin(UUID playerId, int fieldSize, List<Move> moves) {
+        int[] rows = new int[fieldSize];
+        int[] columns = new int[fieldSize];
+        AtomicInteger mainDiagonal = new AtomicInteger();
+        AtomicInteger antiDiagonal = new AtomicInteger();
+
+        return moves.stream().filter(
+                move -> move.getPlayerId().equals(playerId)
+        ).anyMatch(move -> {
+            rows[move.getFirstCoordinate()]++;
```

Comment:
could you extract this to separate method with clear name?

Reply:
I honestly don't see a lot of reason to extract a separate method here. The method is not so big, It's 20 lines long, and It's pretty direct: calculate sums on rows, columns and diagonals, and if any reaches field size, finish. I refactored it as good as I can and I think now it's good enough.

Reply:
the reason why i suggest this is that i didn't get what this methods is doing:)